### PR TITLE
[TDP-16] Spike trap

### PR DESCRIPTION
**Pull request by funkychicken493**
**DEPENDS ON TDP-19 BEING ACCEPTED**

Adds in a spike trap scene that instantly kills the player. Reminder that you can press R to respawn.

### CODE REVIEW

File: player.gd [hunk 1]
```diff
@@ -147,6 +172,15 @@ func update_crouch_state(_old_state: CrouchState, new_state: CrouchState) -> voi
...
 
 func _physics_process(delta: float) -> void:
+	
+	if Input.is_action_just_pressed("debug_respawn"):
+		respawn()
+	elif Input.is_action_just_pressed("debug_die"):
+		die()
+	
+	if dead:
+		return
```
Comment (abby1332) [inline player.gd]: Is this checking every single frame if the player is dead?

File: spike_trap.gd [hunk 2]
```diff
@@ -0,0 +1,9 @@
+extends Node2D
+class_name SpikeTrap
+
+func _on_body_entered(body: Node2D) -> void:
+	if not body is Player:
+		return
+	var player := body as Player
```
Comment (abby1332) [inline spike_trap.gd:7]: Is the spike constantly checking if the player is touching it?

Comment (abby1332) [reply spike_trap.gd:7]: Should the character automatically respawn after death or should the player have to manually respawn?

**Review by abby1332 — APPROVED** [review]

File: spike_trap.gd (hunk 2 above)
Comment (funkychicken493) [reply spike_trap.gd:7]: Player should have to manually spawn. The spike is not constantly checking for player I think.

**Review by ASlightlyOvergrownCactus — APPROVED** [review]
killer review here 👍

**Pull request merged**